basyx.aasxfileserver/basyx.aasxfileserver-core/src/test/java/org/eclipse/digitaltwin/basyx/aasxfileserver/core/AasxFileServerSuite.java

```diff
@@ -0,0 +1,155 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.digitaltwin.basyx.aasxfileserver.AasxFileServer;
+import org.eclipse.digitaltwin.basyx.aasxfileserver.PackageDescription;
+import org.eclipse.digitaltwin.basyx.core.exceptions.ElementDoesNotExistException;
+import org.junit.Test;
+
+/**
+ * Testsuite for implementations of the AasxFileServerSuite interface
+ * 
+ * @author chaithra
+ *
+ */
+public abstract class AasxFileServerSuite {
+	
+	protected abstract AasxFileServer getAasxFileServer();	
+	
+	private static final List<String> DUMMY_AAS_IDS = new ArrayList<>(Arrays.asList("AAS_ID_1", "AAS_ID_2"));	
+	private static final String DUMMY_FILENAME = "test_file.txt";
+	private static final byte[] byteArray = {65, 66, 67, 68, 69};
+	private static final InputStream DUMMY_FILE = new ByteArrayInputStream(byteArray);	
+		
+	@Test
+	public void getAllAASXPackageIds() {
+		
+		AasxFileServer server = getAasxFileServer();		
+		createDummyAASXPackage(server);		
+		Collection<PackageDescription> packageDescriptions = server.getAllAASXPackageIds();			
+		PackageDescription firstPackage = packageDescriptions.iterator().next();	
+
+		assertGetAllAASXPackageIds(packageDescriptions, firstPackage);
+	}
+	
+	@Test
+	public void getAllAASXPackageIdsEmpty() {
+		
+		AasxFileServer server = getAasxFileServer();	
+		Collection<PackageDescription> packageDescriptions = server.getAllAASXPackageIds();					
+				
+		assertIsEmpty(packageDescriptions);		
+	}
+
+	@Test(expected = ElementDoesNotExistException.class)
+	public void getSpecificNonExistingPackageId() {
+		
+		AasxFileServer server = getAasxFileServer();
+		server.getAASXByPackageId("doesNotExist");
+	}
+
+	@Test
+	public void updateExistingAASXByPackageId() {		
+		
+		List<String> expectedAasIds =  new ArrayList<>(Arrays.asList("AAS_ID_3", "AAS_ID_4"));
+		AasxFileServer server = getAasxFileServer();
+		PackageDescription packageDescription = createDummyAASXPackage(server);		
+		server.updateAASXByPackageId(packageDescription.getPackageId(), expectedAasIds, DUMMY_FILE, DUMMY_FILENAME);
+		
+		assertUpdateAASXpaxkageID(packageDescription, expectedAasIds);
+	}	
+
+	@Test(expected = ElementDoesNotExistException.class)
+	public void updateNonExistingAASXByPackageId() {
+		
+		String packageId = "notExisting";		
+		AasxFileServer server = getAasxFileServer();		
+		server.updateAASXByPackageId(packageId, DUMMY_AAS_IDS, DUMMY_FILE, DUMMY_FILENAME);
+	}	
+	
+	@Test
+	public void getAASXByPackageId() throws ElementDoesNotExistException {
+		
+		AasxFileServer server = getAasxFileServer();
+		PackageDescription packageDescription = createDummyAASXPackage(server);			
+		InputStream actualValue = server.getAASXByPackageId(packageDescription.getPackageId());
+		
+		assertEquals(DUMMY_FILE, actualValue);
+	}
+
+	@Test
+	public void deleteAASXPackageById() {
```

Comment:
Please rename it to deleteAASXByPackageId

Reply:
done